go/controller/workflows.go

```diff
@@ -304,6 +328,79 @@ func (wf *Workflows) GetUserByRefreshTokenHash(
 	return user, nil
 }
 
+func (wf *Workflows) UpdateSession( //nolint:funlen
+	ctx context.Context,
+	user sql.AuthUser,
+	refreshToken string,
+	logger *slog.Logger,
+) (*api.Session, error) {
+	userRoles, err := wf.db.GetUserRoles(ctx, user.ID)
+	if err != nil {
+		return nil, fmt.Errorf("error getting roles by user id: %w", err)
+	}
+	allowedRoles := make([]string, len(userRoles))
+	for i, role := range userRoles {
+		allowedRoles[i] = role.Role
+	}
+
+	accessToken, expiresIn, err := wf.jwtGetter.GetToken(
+		ctx, user.ID, allowedRoles, user.DefaultRole, logger,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error getting jwt: %w", err)
+	}
+
+	var metadata map[string]any
+	if len(user.Metadata) > 0 {
+		if err := json.Unmarshal(user.Metadata, &metadata); err != nil {
+			return nil, fmt.Errorf("error unmarshalling user metadata: %w", err)
+		}
+	}
+
+	if _, err := wf.db.UpdateUserLastSeen(ctx, user.ID); err != nil {
+		return nil, fmt.Errorf("error updating user last seen: %w", err)
+	}
+
+	rt, err := wf.GetRefreshTokenByRefreshTokenHash(
```

Comment:
this is redundant, you already did [here](https://github.com/nhost/hasura-auth/pull/492/files#diff-ff137b2c4c6da732fb67bf2e91090217e46acc4be62ffd7a93ddfbd0fb3164dbR16)

In any case, I think there are too many roundtrips here, I think it might be better to have something like:


```sql
-- name: RefreshRefreshToken :one
WITH refresh_token AS (
    UPDATE auth.refresh_tokens
    SET expires_at = $2
    WHERE refresh_token_hash = $1
    RETURNING id, user_id
)
UPDATE auth.users
    SET last_seen = now()
    WHERE id = refresh_token.user_id
    RETURNING refresh_token.id;
```

This single query should:

1. return `pgx.ErrorNoRows` if the hash is wrong so there would be no need to prior check
2. Update the refresh_token expiration
3. Update user's last_seen

this would reduce the endpoint to require only 2 round trips to the database. You may even try to return the roles and do it in a single round-trip but that's probably fine.